tests/mock_network/mod.rs

```diff
@@ -433,96 +426,118 @@ fn carry_out_parsec_pruning() {
-    let new_node_id = FullId::gen(&mut env.new_rng());
-    node_pause_and_resume(env, nodes, new_node_id)
+    let env = Environment::new(NetworkParams {
+        elder_size: LOWERED_ELDER_SIZE,
+        safe_section_size: LOWERED_ELDER_SIZE + 1,
+    });
+
+    let mut nodes = create_connected_nodes(&env, env.safe_section_size());
+    let paused_state = pause_node_and_poll(&env, &mut nodes);
+
+    add_node_to_section(&env, &mut nodes, &Prefix::default());
+    poll_until(&env, &mut nodes, |nodes| {
+        node_joined(nodes, nodes.len() - 1)
+    });
+    let new_id = *nodes.last().unwrap().id();
+
+    nodes.push(TestNode::resume(paused_state));
+
+    // If the paused node is elder, verify it caught up to the new node joining.
+    if nodes.last().unwrap().inner.is_elder() {
+        poll_until(&env, &mut nodes, |nodes| {
+            nodes.last().unwrap().inner.is_peer_our_member(&new_id)
+        })
+    }
+
+    verify_invariants_for_nodes(&env, &nodes);
 }
 
 #[test]
 fn node_pause_and_resume_during_split() {
-    let env = Environment::new(NODE_PAUSE_AND_RESUME_PARAMS);
+    let env = Environment::new(NetworkParams {
+        elder_size: LOWERED_ELDER_SIZE,
+        safe_section_size: LOWERED_ELDER_SIZE + 1,
+    });
 
-    let mut nodes = create_connected_nodes(&env, env.safe_section_size());
-    let prefix =
-        add_connected_nodes_until_one_away_from_split(&env, &mut nodes, &Prefix::default());
+    let mut nodes = vec![];
+    add_mature_nodes(
+        &env,
+        &mut nodes,
+        &Prefix::default(),
+        env.safe_section_size(),
+        env.safe_section_size(),
+    );
+
+    let paused_state = pause_node_and_poll(&env, &mut nodes);
+    nodes.push(TestNode::resume(paused_state));
+
+    poll_until(&env, &mut nodes, |nodes| {
+        section_split(nodes, &Prefix::default())
+    });
 
-    let new_node_id = FullId::within_range(&mut env.new_rng(), &prefix.range_inclusive());
-    node_pause_and_resume(env, nodes, new_node_id)
+    verify_invariants_for_nodes(&env, &nodes);
 }
 
-// Pause a random node, then add new node with the given id, then resume the paused node and verify
-// everything still works as expected.
-fn node_pause_and_resume(env: Environment, mut nodes: Vec<TestNode>, new_node_id: FullId) {
+// Pauses a random node and poll the network for a while. Returns the paused state.
+fn pause_node_and_poll(env: &Environment, nodes: &mut Vec<TestNode>) -> PausedState {
     let index = env.new_rng().gen_range(0, nodes.len());
-    let paused_id = *nodes[index].id();
+    let id = *nodes[index].id();
     let state = nodes.remove(index).inner.pause().unwrap();
 
-    // Verify the other nodes do not see the node as going offline.
-    poll_and_resend(&mut nodes);
+    // Poll the network for a while and verify the other nodes do not see the node as going offline.
+    let start_time = FakeClock::now();
+    poll_until(&env, nodes, |_| {
+        start_time.elapsed() >= Duration::from_secs(60)
```

Comment:
where this magic 60s come from?

Reply:
It's what the comment above says: "poll the network for a while". 60 seconds is "a while" :)

That said, this should probably be changed to something slightly less than 30 seconds because that's the interval after which the node would be detected as non-responsive (`RESEND_MAX_ATTEMPTS * RESEND_DELAY`). The reason why it still works is that the pause/resume is not fully implemented yet and we don't actually drop the quic-p2p instance when pausing. So the remaining nodes never receive the `UnsentUserMessage` events and thus never trigger the non-responsiveness check. This is something that will definitely have to be addressed when we get to finalizing the pause/resume implementation. 

That said, I'll change the interval already and add some explanation to the code.